### PR TITLE
fix: response for non-existent environment

### PR DESCRIPTION
**Pull request by jonas-hoebenreich**
## What does this PR do?

If you call the `/api/v1/client/<environmentID>/website/sync` endpoint with an invalid environmentID, you currently get an `internal server error 500` back. This PR ensures that you get `404 not found` with a helpful error message instead. This also ensures that the app's error log is not getting spammed.

Fixes # (issue)

## How should this be tested?

- Request e.g. `http://localhost:3000/api/v1/client/cm14weni3000108lc7juy7nof/website/sync?version=2.4.1`

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Review by mattinannt — CHANGES_REQUESTED** [review]
@jonas-hoebenreich thanks for the PR; makes sense to return a proper response here :-) 🚀💪
For consistency reasons I would love to see one additional change:

File: apps/web/app/api/v1/client/[environmentId]/website/sync/route.ts [hunk 1]
```diff
@@ -54,12 +54,12 @@ export const GET = async (
...
-    if (!organization) {
-      throw new Error("Organization does not exist");
+    if (!environment) {
+      return responses.notFoundResponse("Environment", environmentId);
     }
 
-    if (!environment) {
-      throw new Error("Environment does not exist");
+    if (!organization) {
+      throw new Error("Organization does not exist");
```
[the remark this answers is not in the document]
Comment (jonas-hoebenreich) [reply apps/web/app/api/v1/client/[environmentId]/website/sync/route.ts:62]: Yes, I thought about that too. Personally, I would leave it as an error, as each environment is assigned to exactly one product and one organization. So if an environment is found but the product or organization is not, I would see this as an error that is not on the user's side. WDYT?

[the remark this answers is not in the document]
Comment (mattinannt) [reply apps/web/app/api/v1/client/[environmentId]/website/sync/route.ts:62]: Yes, that's right and makes sense :-) let's leave the code as it is then 💪.

**Review by mattinannt — APPROVED** [review]

**Comment by pandeymangg** [comment]
hey! thanks for the changes @jonas-hoebenreich 
We just merged the new app surveys approach to the main in #3024, which now removes the older `/website/sync` endpoint in favour of the new environment sync endpoints (`/website/environment` and `/app/environment`), I have made similar changes to those endpoints as well in #3150.
Closing this PR in favour of the changes made in the new one

**Pull request closed without merging**